src/main/java/ratpack/pac4j/RatpackPac4j.java

```diff
@@ -350,24 +360,25 @@ public static Handler requireAuth(Class<? extends Client> clientType, Authorizer
    * }</pre>
    *
    * @param ctx the handling context
-   * @param clientType the client type to authenticate with
+   * @param clientName the name of the client to authenticate with
    * @return a promise for the user profile, fulfilled if logged in
    */
-  public static Promise<UserProfile> login(Context ctx, Class<? extends Client> clientType) {
-    if (isDirect(clientType)) {
+  public static Promise<UserProfile> login(Context ctx, String clientName) {
```

Comment:
I'm not familiar with the usages of this.  Over all, this PR looks like general upgrade work.  The biggest thing that stood out for me was the changing from using "clientType" to "clientName" and some of the 'isDirect(..)'  functional change.

But, I couldn't say what is "right". Seems ok.

Reply:
Yeah, the client name is something that was forced by pac4j itself, there are no longer getters for the type.